pkg/validation/variables/variables.go

```diff
@@ -42,6 +42,9 @@ func ValidateAndReplaceGlobalVariable(workspaceTemplateSpec *v1alpha2.DevWorkspa
 
 		// Validate the starter projects and replace for global variable
 		variableWarning.StarterProjects = ValidateAndReplaceForStarterProjects(workspaceTemplateSpec.Variables, workspaceTemplateSpec.StarterProjects)
+
+		// Validate the starter projects and replace for global variable
+		variableWarning.StarterProjects = ValidateAndReplaceForProjects(workspaceTemplateSpec.Variables, workspaceTemplateSpec.DependentProjects)
```

Comment:
should this be dependentProjects instead?

Reply:
Woops! Yes, good catch. I've updated the PR (and the relevant test case to include dependentProjects)